src/main/java/com/spotify/github/v3/repos/Branch.java

```diff
@@ -52,3 +61,46 @@
   /** Branch protection API URL */
   Optional<URI> protectionUrl();
 }
+
+class BranchDeserializer extends JsonDeserializer<ImmutableBranch> {
```

Comment:
Could BranchDeserializer class be defined in a separate file (BranchDeserializer.java)? 
I understand it compiles just fine and it is a matter of taste... The current code within BranchDeserializer does not look like a data structure definition for example, but rather a utility class.
The code might be cleaner if class is defined within its own java file and it would allow for reusability (if needed).
This is just a comment.